src/utils/symbol_utils.mli

```diff
@@ -14,6 +14,28 @@ type symbol = {
   ; name : string;
 }
 
+(** This type represents an external symbol. *)
+type extern_symbol = {
+  tid : Bap.Std.tid
+  ; address : string
+  ; name : string
+  ; cconv : string option
+  ; args : (Bap.Std.Var.t * Bap.Std.Exp.t * Bap.Std.intent option) list;
+}
+
+
+(** Returns the calling convention for the whole project inferred by Bap. *)
+val get_project_calling_convention : Bap.Std.Project.t -> string option
+
+(** Checks whether the external symbols have already been built. If not, it calls the symbol builder. *)
+val build_and_return_extern_symbols : Bap.Std.Project.t -> Bap.Std.program Bap.Std.term -> extern_symbol list
+
+(** Builds a list of function symbols type from external function names given by objdump. *)
+val build_extern_symbols : Bap.Std.Project.t -> Bap.Std.program Bap.Std.term -> unit
+
+(** Adds an analysed internal symbol to the list of external symbols. *)
+val add_extern_symbol : Bap.Std.Project.t -> Bap.Std.program Bap.Std.term -> string -> unit
```

Comment:
I think the function name is slightly misleading. Maybe rename it in `add_as_extern_symbol` or `mark_as_extern_symbol`